core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1636,6 +1637,10 @@ private void restoreFile(final BlobStoreIndexShardSnapshot.FileInfo fileInfo, fi
                 } else {
                     stream = new RateLimitingInputStream(partSliceStream, restoreRateLimiter, restoreRateLimitingTimeInNanos::inc);
                 }
+
```

Comment:
Thanks @s1monw and @imotov ... I agree we should try to prune such unref'd files before restoring ... seems scary to have mixed alternative future files in the index.  I'll open a follow on issue.